[app] Android 14 でdebug ビルドすると、アプリがクラッシュする

## 概要
下記で起票されているものと同様の事象が起きているので、何か対策出来ないかを考える。

* https://github.com/willowtreeapps/Hyperion-Android/issues/262
* https://github.com/willowtreeapps/Hyperion-Android/issues/264
* https://github.com/willowtreeapps/Hyperion-Android/issues/267



## 作業内容
### 完了条件
* 対策を考え、修正する

### 想定される範囲
* 現時点では不明

### 対応しないこと
* 現時点では特になし



## 備考
### クラッシュログ
<details>
<summary>Android 14 Emulator での実行ログ</summary>

``` log
FATAL EXCEPTION: main
Process: jp.co.yumemi.android.codecheck.debug, PID: 4334
java.lang.RuntimeException: Unable to create service com.willowtreeapps.yperion.core.internal.HyperionService: java.lang.SecurityException: jp.co.umemi.android.codecheck.debug: One of RECEIVER_EXPORTED or ECEIVER_NOT_EXPORTED should be specified when a receiver isn't being egistered exclusively for system broadcasts
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4664)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2264)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.SecurityException: jp.co.yumemi.android.codecheck.debug: ne of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a eceiver isn't being registered exclusively for system broadcasts
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
	at android.os.Parcel.createException(Parcel.java:3041)
	at android.os.Parcel.readException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:2966)
	at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature(IActivityManager.java:5684)
	at java.lang.reflect.Method.invoke(Native Method)
	at leakcanary.ServiceWatcher$install$4$2.invoke(ServiceWatcher.kt:93)
	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
	at $Proxy3.registerReceiverWithFeature(Unknown Source)
	at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1852)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1792)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1780)
	at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:755)
	at com.willowtreeapps.hyperion.core.internal.HyperionService.onCreate(HyperionService.java:114)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4651)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2264) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActivityManagerService.registerReceiverWithFeature(ActivityManagerService.java:13927)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2570)
	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2720)
	at android.os.Binder.execTransactInternal(Binder.java:1339)
	at android.os.Binder.execTransact(Binder.java:1275)
```
</details>

### 試したこと
* AndroidManifest で`FOREGROUND_SERVICE_MEDIA_PROJECTION` を許可してみた → 効果なし
 
### 参考文献
* 現時点では特になし

--- a/app/README.md
+++ b/app/README.md
@@ -10,7 +10,7 @@
 --- | :---: | :---: | ---
 Firebase Crashlytics | - | 利用可能 |  |
 google-services.json | - | 必須 |  |
-[Hyperion] | 利用可能 | - | アプリ起動後に、通知権限を許可した状態でOS 通知欄をタップするか、端末を振るとメニューが出てきます
+[Hyperion] | 利用可能<br />(※Android 13 以下) | - | アプリ起動後に、通知権限を許可した状態でOS 通知欄をタップするか、端末を振るとメニューが出てきます
 [LeakCanary](https://square.github.io/leakcanary/) | 利用可能 | - | アプリ起動後にメモリリークが起きると通知されます
 OkHttp の通信ログ | 利用可能 | - | アプリ起動後にAndroid Studio の"Logcat" ウィンドウを確認してください
 StrictMode | 適用済み | - | 設定内容は[DebugApplication](./src/debug/kotlin/jp.co.yumemi.android.code_check/DebugApplication.kt) を参照してください

--- a/app/src/debug/kotlin/jp/co/yumemi/android/code_check/DebugApplication.kt
+++ b/app/src/debug/kotlin/jp/co/yumemi/android/code_check/DebugApplication.kt
@@ -1,7 +1,9 @@
 package jp.co.yumemi.android.code_check
 
+import android.os.Build
 import android.os.StrictMode
 import androidx.fragment.app.strictmode.FragmentStrictMode
+import com.willowtreeapps.hyperion.core.Hyperion
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 import timber.log.Timber
@@ -53,5 +55,11 @@ class DebugApplication : MainApplication() {
             .build()
             .also { FragmentStrictMode.defaultPolicy = it }
         // endregion
+
+
+        // TODO: Android 14 以降でHyperion が使えるようになったら、記述を削除する
+        if (Build.VERSION_CODES.UPSIDE_DOWN_CAKE <= Build.VERSION.SDK_INT) {
+            Hyperion.disable()
+        }
     }
 }

